### PR TITLE
gh-100414: Skip test_dbm_sqlite3 if sqlite3 is unavailable

### PR DESCRIPTION
**Pull request by erlend-aasland**
Alternative to #115448

* Issue: gh-100414

### CODE REVIEW

**Comment by erlend-aasland** [comment]
!buildbot .*RHEL7.*
!buildbot .*FreeBSD.*

**Comment by erlend-aasland** [comment]
!buildbot .*FreeBSD.*

**Comment by erlend-aasland** [comment]
I think I prefer this fix.

**Review by AlexWaygood — APPROVED** [review]
See my comments in https://github.com/python/cpython/pull/115448#issuecomment-1943626003. This feels slightly more fragile to me, in that whether or not the test passes depends on the order the imports are in, which isn't necessarily obvious to the reader of the test. But either is probably fine; no strong preference :)

**Comment by AlexWaygood** [comment]
> This feels slightly more fragile to me, in that whether or not the test passes depends on the order the imports are in, which isn't necessarily obvious to the reader of the test.

You could maybe add a comment noting that the imports need to be in precisely that order

**Comment by erlend-aasland** [comment]
> You could maybe add a comment noting that the imports need to be in precisely that order

Yes, that seems fair to the future reader.

File: Lib/test/test_dbm_sqlite3.py [hunk 1]
```diff
@@ -7,8 +6,8 @@
 from pathlib import Path
 from test.support import cpython_only, import_helper, os_helper
 
-
 dbm_sqlite3 = import_helper.import_module("dbm.sqlite3")
+import sqlite3
```
Comment (erlend-aasland) [inline Lib/test/test_dbm_sqlite3.py:14]:
```suggestion
# If the import helper managed to import dbm.sqlite3, we must
# inevitably be able to import sqlite3. Else, we have a problem.
import sqlite3
```

Comment (erlend-aasland) [reply Lib/test/test_dbm_sqlite3.py:14]: Something like this, @AlexWaygood?

Comment (AlexWaygood) [reply Lib/test/test_dbm_sqlite3.py:14]: Or maybe:

```suggestion
# N.B. The test will fail if the sqlite3 import is above the import of dbm.sqlite3.
# This is deliberate: if the import helper managed to import dbm.sqlite3,
# we must inevitably be able to import sqlite3. Else, we have a problem.
import sqlite3
```

Comment (erlend-aasland) [reply Lib/test/test_dbm_sqlite3.py:14]: Thanks!

**Comment by erlend-aasland** [comment]
Thanks for the review(s), Alex!

**Comment by AlexWaygood** [comment]
No worries!

**Review by serhiy-storchaka — APPROVED** [review]

**Pull request merged**